### PR TITLE
Enable `compiler_param_file` in Windows MSVC toolchain

### LINKED ISSUE #5163: Compile Line Length Limit hit under Windows
### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

The following build file does not compile under windows
```
cc_library(
    name = "long",
    includes = ["a" * 32768]
)
cc_binary(
    name = "long_bin",
    srcs = ["main.cpp"],
    deps = [
        ":long"
    ]
)
```

The reported error is
`
Action failed to execute: java.io.IOException: ERROR: src/main/native/windows/processes-jni.cc(161): CreateProcessWithExplicitHandles("C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin\amd64\cl.exe" /c main.cpp /Fobazel-out/x64_windows-fastbuild/bin/_objs/long_bin/main.o /nologo /DCOMPILER_MSVC /DNOMINMAX /D_WIN32_WINNT=0x0(...)): command is longer than CreateProcessW's limit (32768 characters)
`

From what I could tell when trying to make a simple repo it appears this case may be handled correctly for link steps but not for compile steps.

### What operating system are you running Bazel on?
Windows 10

### What's the output of `bazel info release`?
release 0.13.0

### PR DESCRIPTION
**Pull request by fmeum**
Fixes #5163

### CODE REVIEW

**Comment by fmeum** [comment]
@oquenchil The tests fail as they inspect the logs and search for compiler flags. Should I make them look into the `param` file or, probably better, inspect `aquery` output instead?

File: tools/cpp/windows_cc_toolchain_config.bzl [hunk 1]
```diff
@@ -377,6 +377,7 @@ def _impl(ctx):
 
         compiler_param_file_feature = feature(
             name = "compiler_param_file",
```
Comment (diegohavenstein) [inline tools/cpp/windows_cc_toolchain_config.bzl:379]: is `linker_param_file` enabled as well?

**Comment by oquenchil** [comment]
> @oquenchil The tests fail as they inspect the logs and search for compiler flags. Should I make them look into the `param` file or, probably better, inspect `aquery` output instead?

I was going to suggest that usually the way we do this is to get to the parameter file writing action and look at the flags there instead. Such a test would be easily converted to a Starlark test. But specifically for compiler_param_file there isn't a separate action, so as you are suggesting, we need a strategy that is part of an integration test.

Do these compiler flags show up in aquery? If they do, that's probably better than reading the parameter file since it avoids execution.

File: tools/cpp/windows_cc_toolchain_config.bzl (hunk 1 above)
Comment (oquenchil) [reply tools/cpp/windows_cc_toolchain_config.bzl:379]: Having a quick look it seems nothing has changed with respect to the feature linker_param_file in a long time. You can try passing --features=linker_param_file, if that works great. It still wouldn't explain why you see this difference with respect to 5.4.

Let's just focus on compiler_param_file feature here and let's continue on https://github.com/bazelbuild/bazel/issues/17068.

**Comment by oquenchil** [comment]
Thank you for the very quick PR Fabian.

**Comment by fmeum** [comment]
@oquenchil Tests are passing now. I also added a test for long command lines based on your gist.

**Review by oquenchil — APPROVED** [review]
Thank you Fabian!

**Pull request closed without merging**